src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -790,6 +790,123 @@ public StringBuilder Append(string value, int startIndex, int count)
             }
         }
 
+        public StringBuilder Append(StringBuilder value)
+        {
+            if (value != null && value.Length != 0)
+            {
+                if (value != this)
+                    AppendCore(value, 0, value.Length);
+                else
+                    Append(value.ToString());
+            }
+            return this;
+        }
+
+        public StringBuilder Append(StringBuilder value, int startIndex, int count)
+        {
+            if (startIndex < 0)
```

Comment:
Please use ThrowHelper instead of newing up an exception inline. See elsewhere. Probably the existing code in this class should use ThrowHelper, at least in the "hot" methods (maybe you could do that in another commit)

See reasoning for this:
https://github.com/dotnet/corefx/issues/26253#issuecomment-356736809

Reply:
Should I change this for existing append overloads ?

Reply:
We do the manual optimization using ThrowHelper in hot small methods or generic methods because of it is where it matters the most. We do not do it in "regular" methods like this because of it is extra pain to maintain.

Hopefully, we will eventually get optimization step in the build process that does the transformation of throw to ThrowHelper everywhere and we will not need to ever worry about it again.

Reply:
You may want to undo the use of throw helper here to be consistent with the overall style. These methods are neither generic or hot.